src/core/Transcriber.py

```diff
@@ -1,90 +1,241 @@
+                    
+                    if self.transcription_process.is_alive():
+                        logging.warning(f"Process didn't terminate gracefully, killing it")
+                        self.transcription_process.kill()
+                        self.transcription_process.join()
+                    
+                    self._handle_cancellation()
+                    return
+                
+                # Check every 0.5 seconds
+                time.sleep(0.5)
+            
+            # Process completed, get result
+            self.transcription_process.join()
+            
+            # Check if process was successful
+            if not result_queue.empty():
+                result_data = result_queue.get()
+                
+                if result_data["success"]:
+                    result = result_data["result"]
+                    
+                    # Store results
+                    self.whisper_result = result
+                    self.ts_api.database.update_job_status(
+                        self.module_entry.uid,
+                        JobStatus.COMPLETED,
+                        whisper_result=result,
+                        completed_at=time.time()
+                    )
+                    
+                    # Clean up audio file
+                    if os.path.exists(self.file_path):
+                        os.remove(self.file_path)
+                        logging.debug(f"Removed audio file for job {self.module_entry.uid}")
+                    
+                    logging.info(f"Successfully completed job {self.module_entry.uid}")
+                else:
+                    raise Exception(result_data["error"])
+            else:
+                # Process exited without result
+                raise Exception("Transcription process exited without result")
+            
         except Exception as e:
-            logging.error(e)
-            self.ts_api.database.change_job_entry(self.module_entry.uid,
-                                                  "status", 4)  # Failed
-            self.ts_api.unregister_job(self.module_entry)
+            logging.error(f"Error transcribing job {self.module_entry.uid}: {e}")
+            self.ts_api.database.update_job_status(
+                self.module_entry.uid,
+                JobStatus.FAILED,
+                error_message=str(e),
+                completed_at=time.time()
+            )
+            
+            # Clean up audio file on error
+            if os.path.exists(self.file_path):
+                try:
+                    os.remove(self.file_path)
+                except:
+                    pass
+            
+            # Clean up process if still running
+            if self.transcription_process and self.transcription_process.is_alive():
+                self.transcription_process.terminate()
+                self.transcription_process.join(timeout=2)
+    
```

Comment:
Similarly how it is done in line [148](https://github.com/uni-halle/ts-api/blob/52dcfd7a91456298fb1a13e3fe504f9126b2ada8/src/core/Transcriber.py#L148) we can add a possibility of the programme to kill the process if ending it gracefully won't work.
```suggestion
            if self.transcription_process.is_alive():  
                logging.warning(f"Process didn't terminate gracefully, killing it")  
                self.transcription_process.kill()  
                self.transcription_process.join()
```